src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplesController.java

```diff
@@ -286,16 +287,24 @@ public String getSampleFiles(final Model model, @PathVariable Long projectId, @P
 		Sample sample = sampleService.read(sampleId);
 		model.addAttribute("sampleId", sampleId);
 
-		Collection<SampleSequencingObjectJoin> filePairJoins = sequencingObjectService
-				.getSequencesForSampleOfType(sample, SequenceFilePair.class);
-		Collection<SampleSequencingObjectJoin> singleFileJoins = sequencingObjectService
-				.getSequencesForSampleOfType(sample, SingleEndSequenceFile.class);
+		Collection<SampleSequencingObjectJoin> filePairJoins = sequencingObjectService.getSequencesForSampleOfType(
+				sample, SequenceFilePair.class);
+		Collection<SampleSequencingObjectJoin> singleFileJoins = sequencingObjectService.getSequencesForSampleOfType(
+				sample, SingleEndSequenceFile.class);
+		Collection<SampleSequencingObjectJoin> fast5FileJoins = sequencingObjectService.getSequencesForSampleOfType(
+				sample, Fast5Object.class);
+		List<Fast5Object> fast5 = fast5FileJoins.stream()
+				.map(j -> (Fast5Object) j.getObject())
```

Comment:
No need to cast here.  You're not doing anything with them after they're cast. It cna just be a List of `SequencingObject` I think.

Reply:
Updated in 0d74082a0fff036d27cdfcfa1007c91abbd9cee6